lib/vtk/commands/socks/setup.rb

```diff
@@ -0,0 +1,327 @@
+          @port = options['port'] || '2001'
+          @boot_script_path = options['boot_script_path'] || "#{ENV['HOME']}/Library"
+          @ssh_key_path = options['ssh_key_path'] || "#{ENV['HOME']}/.ssh/id_rsa_vagov"
+          @ssh_config_path = options['ssh_config_path'] || "#{ENV['HOME']}/.ssh/config"
+          @skip_test = options['skip_test'] || false
+
+          super()
+        end
+
+        def execute(input: $stdin, output: $stdout)
+          @input = input
+          @output = output
+
+          check_ssh_key
+          configure_ssh
+          test_ssh_connection unless skip_test
+
+          system_boot_configured = configure_system_boot
+          configure_system_proxy
+
+          test_http_connection if system_boot_configured && !skip_test
+
+          log 'SOCKS setup complete.'
+          log 'Please run `vtk socks on` to start your SOCKS connection.' unless system_boot_configured
+        end
+
+        private
+
+        def check_ssh_key
+          return true if key_exists?
+
+          open_key_access_request
+          exit
+        end
+
+        def key_exists?
+          File.exist? ssh_key_path
+        end
+
+        def open_key_access_request
+          url = 'https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?' \
+            'assignees=&labels=external-request%2C+operations&template=Environment-Access-Request-Template.md&' \
+            'title=Access+for+%5Bindividual%5D'
+          log "Please create an SSH key using `ssh-keygen -f ~/.ssh/id_rsa_vagov`. You'll have to wait for access " \
+            'approval before continuing. Once approved, re-run `vtk socks setup`.'
+          `#{macos? ? 'open' : 'xdg-open'} "#{url}"` if prompt.yes? 'Open access request form in GitHub?'
+        end
+
+        def configure_ssh
+          install_ssh_config
+          configure_ssh_config_with_keychain
+          ssh_config_clean_up
+          ssh_agent_add
+        end
+
+        def install_ssh_config
+          return true if ssh_config_configured?
+          return true unless prompt.yes? "----> #{pretty_ssh_config_path} missing or incomplete. Install/replace now?"
+
+          ssh_dir = File.dirname ssh_config_path
+
+          download_ssh_config unless File.exist? '/tmp/dova-devops'
+          FileUtils.mkdir_p ssh_dir
+          FileUtils.chmod 0o700, ssh_dir
+          backup_existing_ssh_config
+          FileUtils.cp '/tmp/dova-devops/ssh/config', ssh_config_path
+          FileUtils.chmod 0o600, "#{ssh_dir}/config"
+        end
+
+        def ssh_config_configured?
+          return false unless ssh_config_exists?
+
+          download_ssh_config
+          ssh_config_local = File.read ssh_config_path
+          ssh_config = File.read '/tmp/dova-devops/ssh/config'
+          ssh_config_local.include? ssh_config
+        end
+
+        def ssh_config_exists?
+          File.exist? ssh_config_path
+        end
+
+        def download_ssh_config
+          install_brew
```

Comment:
Brew is installed here as a new machine won't have Xcode Tools which is need to use git.